src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -1343,12 +1343,12 @@ private JwtSecurityToken ValidateSignature(string token, JwtSecurityToken jwtTok
             bool kidMatched = false;
             IEnumerable<SecurityKey> keys = null;
 
+            if (validationParameters.RequireSignedTokens == false)
```

Comment:
The logic was if there was a signature, we checked it if the flag was true (default).
With this change, we will not check it.

Reply:
Does it mean it's not suitable?

Reply:
I'mnot sure if I understand the comment, did I make a mistake here? 
What I was going after was to not check signature at all if the flag is set to false and otherwise check it fully...  does that assumption make sense?

Reply:
@jmprieur @MZOLN i am calling out a behavior change where we always checked the signature if it existed.  Customers that had set the flag to false by mistake, would still have the signature validated if it existed.

Reply:
OH I understand now. Thank you for the call out.

The reason for the change is because customers who set this flag purposefully, e.g. for testing, have a hard time figuring out how to use it. Currently you have to remove the signature part of the token and leave the trailing "." behind (for the split to work). Their expectation is that all signature validation is bypassed. Maybe another flag would be a  better solution?

Reply:
@MZOLN that is why the flag says - Require...., rather than Skip...

Reply:
@MZOLN i don't think we should make this change.